web/client/components/map/leaflet/plugins/VectorLayer.jsx

```diff
@@ -40,16 +40,28 @@ var createVectorLayer = function(options) {
         style: options.nativeStyle || options.style || defaultStyle // TODO ol nativeStyle should not be taken from the store
     });
     layer.setOpacity = (opacity) => {
-        const style = assign({}, layer.options.style || defaultStyle, {opacity: opacity, fillOpacity: opacity});
+        const style = assign({}, layer.options.style || defaultStyle, { opacity: opacity, fillOpacity: opacity });
         layer.setStyle(style);
         setOpacity(layer, opacity);
     };
+    layer.on('layeradd', () => {
+        layer.setOpacity(layer.opacity || options.opacity);
```

Comment:
opacity is a number, so it can be 0. Please make it work also for 0 value, and provide a test for this.